lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -97,6 +88,22 @@ class GoogleCloudOutput < BufferedOutput
     #   }
     config_param :label_map, :hash, :default => nil
 
+    # Credentials and Authorization
+    #
+    # Method 1: Use Application Default Credentials / GCE service account
+    #
+    # This method makes use of the application default credentials. These
+    # are provided by GCE, or through the the 'gcloud auth' command.
+    # This requires no parameters, though it may read credentials
+    # located by the environment variable GOOGLE_APPLICATION_CREDENTIALS.
+    #
+    # Method 2: Use Service account keys.
+    #
+    # service_credentials_path is a path to a service account key file,
+    # in json format, which can be acquired from the Google Cloud Developers
+    # Console. When set, the other fields are unnecessary.
+    config_param :service_credentials_path, :string, :default => nil
```

Comment:
Yeah, I went down that path, and ran into a problem.

Specifically, the credentials needed for my next piece are not a part of default application credentials. So users must install their own.

So, I can either reuse the default application credentials, or I can provide a config for them.

Overriding default application credentials seems a bit obscure and error-prone. Ok, so users can download credentials and store them in a separate path. Then I have to provide a method to change the appropriate environment variables. That is essentially equivalent to the config, however it is awkward to accomplish when starting a service. Documentation and explanation for this case would become quite complex.

Suppose, also, that the user wants to run my new config and logging. Now they may need to provide two different sets of credentials. Yet there is only a single mechanism to provide credentials. :/

I take issue with the design of google default application credentials library; essentially it requires all parameters passed as env variables, and it does not provide good mechanisms to get the credentials in other ways. But that is neithere here nor there.